Projects/Server/Gumps/New/Components/Interpolation/GumpInterpolatedStringHandler.cs

```diff
@@ -0,0 +1,125 @@
+/*************************************************************************
+ * ModernUO                                                              *
+ * Copyright 2019-2023 - ModernUO Development Team                       *
+ * Email: hi@modernuo.com                                                *
+ * File: GumpInterpolatedStringHandler.cs                                *
+ *                                                                       *
+ * This program is free software: you can redistribute it and/or modify  *
+ * it under the terms of the GNU General Public License as published by  *
+ * the Free Software Foundation, either version 3 of the License, or     *
+ * (at your option) any later version.                                   *
+ *                                                                       *
+ * You should have received a copy of the GNU General Public License     *
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>. *
+ *************************************************************************/
+
+using Server.Gumps.Interfaces;
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+using static Server.Gumps.Components.Interpolation.GumpInterpolatedStringHandler;
+
+namespace Server.Gumps.Components.Interpolation
+{
+    [InterpolatedStringHandler]
+    public ref struct GumpInterpolatedStringHandler<TStringHandler, TFormatter>
+        where TStringHandler : struct, IStringsHandler
+        where TFormatter : struct, IGumpInterpolationTextFormatter<TFormatter>
+    {
+        private static readonly char[] _buffer = GC.AllocateUninitializedArray<char>(1024);
```

Comment:
I think this gets replicated for reach concrete type. At least that is how it works for classes. We have to find out if it works the same with ref structs (since that is a new construct), or if it is done by erasure.

Reply:
Moved outside the generic class into file scoped static class